src/core.js

```diff
@@ -0,0 +1,661 @@
+        var data = new Uint8Array(3);
+        data[0] = youme.write.statusByte(msNibble, lsNibble);
+        data[1] = data1;
+        data[2] = data2;
+        return data;
+    };
+
+    // Unsupported, non-API function.
+    youme.write.largeValueMessage = function (msNibble, lsNibble, midiMessage) {
+        var data = new Uint8Array(3);
+        data[0] = youme.write.statusByte(msNibble, lsNibble);
+        youme.write.twoByteValue(midiMessage.value, data, 1);
+        return data;
+    };
+
+    /**
+     *
+     * Output a status byte.
+     *
+     * @param {Number} msNibble - the first nibble of the status byte (often the command code).
+     * @param {Number} lsNibble - the second nibble of the status byte (often the channel).
+     * @return {Byte} A status byte that combines the two inputs.
+     */
+    // Unsupported, non-API function.
+    youme.write.statusByte = function (msNibble, lsNibble) {
+        return (msNibble << 4) + lsNibble;
+    };
+
+    /**
+     *
+     * Converts a 14-bit numeric value to two MIDI bytes.
+     *
+     * @param {Number} value - A 14-bit number to convert
+     * @param {Unit8TypedArray} array - An array to write the value to.
+     * @param {Integer} offset - The optional offset in the array to start writing at.  Defaults to 0.
+     *
+     */
+    // Unsupported, non-API function.
+    youme.write.twoByteValue =  function (value, array, offset) {
+        offset = offset || 0;
+        array[offset] = value & 0x7f; // LSB
+        array[offset + 1] = (value >> 7) & 0x7f; // MSB
+    };
+
+    /**
+     *
+     * Convert a MIDI Message represented as a Javascript Object into a Sysex message represented as a Uint8Array.
+     *
+     * NOTE: This function does not accept framing, i.e. a leading 0xF0 and/or trailing 0xF7, and will fail if called
+     * with either.
+     *
+     * @param {Object} midiMessage - The MIDI message represented as a Javascript Object.
+     * @return {Uint8Array} - The sysex message.
+     */
+    // Unsupported, non-API function.
+    youme.write.sysex = function (midiMessage) {
+        if (midiMessage.data[0] === 0xF0 || midiMessage.data[midiMessage.data.length - 1] === 0xF7) {
+            fluid.fail("Sysex payloads should not include framing bytes.");
+        }
+
+        var data = midiMessage.data,
+            len = data.length;
+
+        var framedData = new Uint8Array(len + 2);
+        framedData[0] = 0xF0;
+        framedData[len + 1] = 0xF7;
+        framedData.set(data, 1);
+
+        return framedData;
+    };
+
+    /**
+     *
+     * Search a map of available ports for matches.
+     *
+     * @param {Array<MIDIPort>} ports - The ports to search.
+     * @param {PortSpec} portSpec - An object that describes the desired port.
+     * @return {Array<MIDIPort>} - An array of matching inputs.
+     *
+     */
+    youme.findPorts = function (ports, portSpec) {
+        var portFinder = youme.findPorts.portFinder(portSpec);
```

Comment:
This is more a "note to self" than anything else - note that what we would really like to achieve is that portFinders are some kind of "memoised" function of the portSpec. This is the kind of thing that React's programming model makes quite easy. As it is, we have to recreate this garbage on every invocation of this function whether portSpec has changed or not.